tests/kernel/version_header/src/main.c

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2016 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <ztest.h>
+#include <misc/printk.h>
+#include "version.h" /* generated by MAKE, at compile time */
+#include <version_header.h>
+
+void test_version(void)
+{
+#if CONFIG_VERSION_HEADER
+	zassert_true(version_header.major ==
+		     KERNEL_VERSION_MAJOR, "major version mismatch");
+	zassert_true(version_header.minor ==
+		     KERNEL_VERSION_MINOR, "minor version mismatch");
+	zassert_true(version_header.patch
+		     == KERNEL_PATCHLEVEL, "patchlevel version match");
+#endif
```

Comment:
#else 
      zassert_unreachable("CONFIG_VERSION_HEADER not defined");

Please consider adding this #else statement. This will ensure compilation that did not add CONFIG_VERSION_HEADER will failed and not slipping through unexpectedly. If the test should not be run on the platform, then it should be excluded in testcase.yaml